### PR TITLE
[FEATURE] Handle heartbeat from unknown worker

### PR DESCRIPTION
**Pull request by TonyDoen**
# [FEATURE] Handle heartbeat from unknown worker

### What changes were proposed in this pull request?
When Worker heartbeats to Master, the NonFatal exception occurs 5 times in a row that triggers the stop of Worker

### Why are the changes needed?
The changes will try to make sure qualified Worker can be used.

### Related issues.
https://github.com/alibaba/RemoteShuffleService/issues/4

### How was this patch tested?

/cc @FMX

/assign @wangshengjie3

### CODE REVIEW

File: server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Worker.scala [hunk 1]
```diff
@@ -156,14 +158,27 @@ private[deploy] class Worker(
...
+
+      heartBeatFailCnt.getAndSet(0)
+    } catch {
+      case NonFatal(t) =>
+        if (heartBeatFailCnt.addAndGet(1) > 5) {
+          logError(s"Uncaught exception 5 times. Now stopping in thread " +
+              s"${Thread.currentThread().getName}", t)
+          onStop()
+        } else {
+          logError(s"Uncaught exception in thread ${Thread.currentThread().getName}", t)
```
Comment (wangshengjie123) [inline server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Worker.scala:179]: Log is not exact, `Uncaught Exception` ?

Comment (TonyDoen) [reply server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Worker.scala]: thanks, I have changed the log message.

File: server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Worker.scala [hunk 2]
```diff
@@ -156,14 +158,28 @@ private[deploy] class Worker(
...
+        , classOf[HeartbeatResponse])
+      cleanTaskQueue.put(response.expiredShuffleKeys)
+
+      heartBeatFailCnt.getAndSet(0)
+    } catch {
+      case NonFatal(t) =>
+        if (heartBeatFailCnt.addAndGet(1) > 5) {
+          logError(s"Heart-beat fail 5 times in a row. Now stopping in thread " +
+              s"${Thread.currentThread().getName}", t)
+          onStop()
```
Comment (waitinfuture) [inline server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Worker.scala:177]: Is it tested that this make the worker stop?

Comment (waitinfuture) [reply server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Worker.scala:177]: Will the worker be restarted after stop?

**Pull request closed without merging**